### DISCUSSION TITLE
How to run a Let's Encrypt deploy hook only for a single domain?

### DISCUSSION THREAD

**Discussion by Turab** [caprover/caprover#1838] [Q&A]
I am sure I have seen this before but can't find it in docs or anywhere now.

There is already /captain/data/letencrypt/etc/renewal-hooks/deploy but as you know this is global and it runs eveytime a sertificate deploys. I need domain specific hook. I see that it is also achieved through Let's Encrypt CLI but we don't have access to how Caprover runs that command, right?

There is also /etc/letsencrypt/renewal/DOMAIN.conf in which a renewal hook directive could be placed but in Caprover instance, there is no /etc/letsencrypt directory.

So how do we add a hook to only one single domain?

**Comment by githubsaturn** [comment]
You can't edit CapRover renew commands unless you fork and rebuild CapRover. 

I'm not too familiar with the Let'sEncrypt hooks, but I am pretty sure there must be a way that they pass the renewed domain to the hook - either through a argument passed to the hook, or an env var. You can then make your logic check for these before taking any actions.
